src/psyclone/nemo.py

```diff
@@ -0,0 +1,822 @@
+                # and thus we need an explicit loop for this dimension.
+                outermost_dim = idx
+
+        # TODO allow for implicit loops with specified bounds (e.g. 2:jpjm1)
+        self._start = 1
+        self._step = 1
+        if outermost_dim == 0:
+            # TODO ensure no name clash is possible with variables that
+            # already exist in the NEMO source.
+            self._variable_name = name_space_manager.create_name(
+                root_name="psy_ji", context="PSyVars", label="psy_ji")
+            self.loop_type = "lon"
+            self._stop = "jpi"
+        elif outermost_dim == 1:
+            self._variable_name = name_space_manager.create_name(
+                root_name="psy_jj", context="PSyVars", label="psy_jj")
+            self.loop_type = "lat"
+            self._stop = "jpj"
+        elif outermost_dim == 2:
+            self._variable_name = name_space_manager.create_name(
+                root_name="psy_jk", context="PSyVars", label="psy_jk")
+            self.loop_type = "levels"
+            self._stop = "jpk"
+        else:
+            raise GenerationError(
+                "Array section in unsupported dimension ({0}) for code "
+                "'{1}'".format(outermost_dim+1, str(ast)))
+        # TODO Since the fparser2 AST does not have parent
+        # information (and no other way of getting to the root node), it is
+        # currently not possible to insert a declaration in the correct
+        # location.
+        # For the moment, we can work around the fparser2 AST limitation
+        # by using the fact that we *can* get hold of the PSyclone Invoke
+        # object and that contains a reference to the root of the fparser2
+        # AST...
+        name = Fortran2003.Name(FortranStringReader(self._variable_name))
+        prog_unit = self.root.invoke._ast
+        spec = walk_ast(prog_unit.content, [Fortran2003.Specification_Part])
+        if not spec:
+            names = walk_ast(prog_unit.content, [Fortran2003.Name])
+            # TODO create a Specification_Part
+            raise InternalError("No specification part found for routine {0}!".
+                                format(names[0]))
+        # TODO check that this variable has not already been declared
+        # Requires that we capture all variable declarations in the routine
+        # in some sort of management class.
+        decln = Fortran2003.Type_Declaration_Stmt(
+            FortranStringReader("integer :: {0}".format(self._variable_name)))
+        spec[0].content.append(decln)
+
+        for subsec in subsections:
+            # A tuple is immutable so work with a list
+            indices = list(subsec.items)
+            # Replace the colon with our new variable name
+            indices[outermost_dim] = name
+            # Replace the original tuple with a new one
+            subsec.items = tuple(indices)
+
+        # Create an explicit loop
+        text = '''\
```

Comment:
It's a small thing but I prefer using
(
"xxx\n"
"yyy\n"
)
as it allows us to inline the code more nicely.

Reply:
Done.